src/dmclock_server.h

```diff
@@ -1762,13 +1762,13 @@ namespace crimson {
 
 	while (!this->finishing) {
 	  // predicate for cond.wait()
-	  const auto pred = [this] () -> bool { return this->finishing; };
+	  const auto pred = [this] () -> bool { return this->finishing || sched_ahead_when > TimeZero; };
 
 	  if (TimeZero == sched_ahead_when) {
 	    sched_ahead_cv.wait(l, pred);
 	  } else {
 	    // cast from Time -> duration<Time> -> Duration -> TimePoint
-	    const auto until = typename super::TimePoint{
+	    const auto until = typename std::chrono::system_clock::time_point{
```

Comment:
@yongseokoh could you elaborate a little bit why we need to have this change? i think any timepoint should work here as long as `chrono::is_clock_v<TimePoint::clock>. as the standard library should convert it to `steady_clock` or `system_clock`.

Reply:
@tchaikov that's a good point. I will revert the line.